File: Data Structures & Algorithms/remove-element/submission-3.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        vector<int>ans;
        int count = 0;
        for (int idx = 0; idx<nums.size(); idx++) {
            if (nums[idx] != val) {
                ans.push_back(nums[idx]);
                count++;
            }
        }
        nums = ans;
        return count;
    }
};

File: Data Structures & Algorithms/majority-element-ii/submission-3.rs
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> Vec<i32> {
        let n = nums.len();
        let mut checked = HashSet::new();
        let mut res = vec![];

        for idx in 0..n {
            let num = nums[idx];
            if checked.contains(&num) {
                continue;
            } else {
                let count = nums.iter().filter(|&&val| val == num).count();
                if count > n / 3 {
                    res.push(num);
                }
                checked.insert(num);
            }
        }

        res
    }
}